boa/src/value/mod.rs

```diff
@@ -925,11 +934,100 @@ impl Value {
 
     #[inline]
     pub fn to_property_descriptor(&self, context: &mut Context) -> Result<PropertyDescriptor> {
-        if let Self::Object(ref object) = self {
-            object.to_property_descriptor(context)
```

Comment:
I think its best to keep a `to_property_descriptor()` function in `GcObject` too, and call it from here when it is an object and throw otherwise. This is a small deviation from spec (not in behaviour), but it avoids some corner cases where we have an object and we want to turn it into a property descriptor, we would have to do `Value::from(object).to_property_descriptor()`. So we can avoid an unnecessary conversion and check if it is an object.

Reply:
Moved `to_property_descriptor` to `GcObject`, preserving the check on `Value` in case someone has to call that function on an unwrapped `Object`.